application_sandbox/private_data/main.cpp

```diff
@@ -0,0 +1,360 @@
+// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+
+#include "application_sandbox/sample_application_framework/sample_application.h"
+#include "mathfu/matrix.h"
+#include "mathfu/vector.h"
+#include "support/entry/entry.h"
+#include "vulkan_helpers/buffer_frame_data.h"
+#include "vulkan_helpers/helper_functions.h"
+#include "vulkan_helpers/vulkan_application.h"
+#include "vulkan_helpers/vulkan_model.h"
+
+using Mat44 = mathfu::Matrix<float, 4, 4>;
+using Vector4 = mathfu::Vector<float, 4>;
+
+namespace cube_model {
+#include "cube.obj.h"
+}
+const auto& cube_data = cube_model::model;
+
+uint32_t cube_vertex_shader[] =
+#include "cube.vert.spv"
+    ;
+
+uint32_t cube_fragment_shader[] =
+#include "cube.frag.spv"
+    ;
+
+struct CubeFrameData {
+  containers::unique_ptr<vulkan::VkCommandBuffer> command_buffer_;
+  containers::unique_ptr<vulkan::VkFramebuffer> framebuffer_;
+  containers::unique_ptr<vulkan::DescriptorSet> cube_descriptor_set_;
+};
+
+static uint64_t kPrivateDataValue = 12345L;
+
+static VkPhysicalDevicePrivateDataFeaturesEXT ext_features{
```

Comment:
I think we should add a VkDevicePrivateDataCreateInfoEXT as part of vkCreateDevice pNext chain to fully exercise the extension. This is important for serialization.

Reply:
Oh I missed it, done, thank you!